### PR TITLE
PTBR Translation

### PR DESCRIPTION
**Pull request by gsantos9489**
Translation for the *.resx files to Brazilian Portuguese (pt-BR).

If you guys find anything wrong, please contact me.

I'm already a software developer, but I never ventured before on open-source projects.
So, this is my first contribution ever. Hope I can contribute more in the future.

Best Regards

### CODE REVIEW

File: GUI/Properties/Resources.pt-BR.resx [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+  <resheader name="reader">
+    <value>System.Resources.ResXResourceReader, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
+  </resheader>
+  <resheader name="writer">
+    <value>System.Resources.ResXResourceWriter, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
+  </resheader>
+  <assembly alias="System.Windows.Forms" name="System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" />
+  <data name="ModuleLabelListFavourites" xml:space="preserve"><value>Favorites</value></data>
+  <data name="EditModpackTooltipLicense" xml:space="preserve"><value>The license for this modpack</value></data>
+  <data name="ModInfoLicenseLabel" xml:space="preserve"><value>License:</value></data>
```
Comment (HebaruSan) [inline GUI/Properties/Resources.pt-BR.resx:123]: I think this is English, and many strings are missing. See the complete list here:

https://github.com/KSP-CKAN/CKAN/blob/master/GUI/Properties/Resources.resx

**Comment by HebaruSan** [comment]
> So, this is my first contribution ever. Hope I can contribute more in the future.

Thanks for helping us out, and we'd love to have more contributions!

I'll be adding some more changes in a few minutes that will allow a test build to be created so you can try out your changes...

**Comment by HebaruSan** [comment]
OK, if you scroll down to the bottom of this page, there's a test build that should support pt-BR:

https://github.com/KSP-CKAN/CKAN/actions/runs/717941813

File: GUI/Properties/Resources.pt-BR.resx (hunk 1 above)
Comment (gsantos9489) [reply GUI/Properties/Resources.pt-BR.resx:123]: Hi,

Will do the missing translations tonight.

Thanks

**Comment by HebaruSan** [comment]
This looks really good! Checking for issues on Linux, we may need to adjust the sizes of some of the controls in the settings...

image

image

image

image

And the compatible versions popup looks like a label doesn't quite fit:

image

**Comment by gsantos9489** [comment]
I think I can fit some of those strings by just finding a better translation.
If not, there is a way to adjust the visual components?

**Comment by HebaruSan** [comment]
> If not, there is a way to adjust the visual components?

Yes, the locations and sizes are all in the `.Designer.cs` file:

https://github.com/KSP-CKAN/CKAN/blob/4e24870813a0efd6179382f3ecb4c24b866b28db/GUI/Dialogs/SettingsDialog.Designer.cs

You don't have to take responsibility for doing that yourself, though. If there are strings that just can't fit, we can help out with trying to shift the layout.

**Comment by gsantos9489** [comment]
Ok, I'll commit some strings in the next few minutes.
I'm kinda new to C#, so if you guys can do that for me, I'll be grateful.

**Comment by gsantos9489** [comment]
I adjusted some strings.
The others really need some component location and sizes attention.

So... what are the next steps?

**Comment by HebaruSan** [comment]
> So... what are the next steps?

It's getting a little hate here, sorry. I'll start fiddling with the layout tomorrow. Thanks for the updates!

**Comment by HebaruSan** [comment]
Updated screenshot for reference:

image

**Comment by HebaruSan** [comment]
With [the latest test build](https://github.com/KSP-CKAN/CKAN/suites/2443070338/artifacts/52313062):

image

FYI our French-speaking friend @vinix38 that more words might fit into `RefreshPreLabel.Text` now ("Rafraîchissement liste:" seems very short).

**Comment by HebaruSan** [comment]
@DasSkelett would you mind taking a look at my commits so I'm not "reviewing" changes that I've also made?

Also we still need to figure out what to do about this label in the compatibility popup. I'm thinking that text box doesn't need to be nearly that wide...

image

**Comment by HebaruSan** [comment]
Coming soon...

image

**Comment by gsantos9489** [comment]
Seems that everything is fine now... If there anything else that I can do.
Please contact me.

**Comment by HebaruSan** [comment]
> Seems that everything is fine now... If there anything else that I can do.

Yeah, the translation does look good, thanks! We just need to make sure I haven't messed something up for the other languages with these changes. Or whatever problem DasSkelett thinks of.

**Comment by DasSkelett** [comment]
The auth token window has some alignment issues:
image

And the text of the "Check for updates"/"Checar por atualizações" button in the settings doesn't quite fit:
image

Also some of the dynamic strings like `MainInstallModSuccess` or `MainModListLoadingAvailable` show up in German for me, but they are translated in `Properties.pt-BR.rsx`, so not sure what's going on there.

**Comment by HebaruSan** [comment]
> And the text of the "Check for updates"/"Checar por atualizações" button in the settings doesn't quite fit:
> image

Good catch, missed that because I didn't know what it was supposed to say :grin: .
Luckily we've got all this extra vertical space in that frame now. Any objections to using it?

image

> Also some of the dynamic strings like `MainInstallModSuccess` or `MainModListLoadingAvailable` show up in German for me, but they are translated in `Properties.pt-BR.rsx`, so not sure what's going on there.
> This also affects other languages though.

I've been seeing the same (but with en-US strings). I wonder if we need to revisit #2899.

**Comment by DasSkelett** [comment]
> Good catch, missed that because I didn't know what it was supposed to say 😁 .
> Luckily we've got all this extra vertical space in that frame now. Any objections to using it?

Go for it!

> I've been seeing the same (but with en-US strings). I wonder if we need to revisit #2899.

I'm gonna try and see if reverting it changes anything.

**Comment by HebaruSan** [comment]
> > Also some of the dynamic strings like `MainInstallModSuccess` or `MainModListLoadingAvailable` show up in German for me, but they are translated in `Properties.pt-BR.rsx`, so not sure what's going on there.
> > This also affects other languages though.
> 
> I've been seeing the same (but with en-US strings). I wonder if we need to revisit #2899.

Some... _interesting_ results printing `resourceFileName` in `SingleAssemblyResourceManager` during startup (this is with pt-BR selected in the dropdown):

image

(and so on.) Guessing that some persistent component gets initialized before we load the language setting?

**Comment by HebaruSan** [comment]
May also be related to cleaning up those `'ResourceManager.ResourceSets' is obsolete` warnings in #3315.
Or we need to set `Thread.CurrentThread.CurrentCulture` in addition to `Thread.CurrentThread.CurrentUICulture`?
Or we need to set `Thread.CurrentThread.CurrentUICulture` for other threads?

**Comment by DasSkelett** [comment]
Reverting #2899 didn't fix it, trying #3315 next.

**Comment by HebaruSan** [comment]
> Or we need to set `Thread.CurrentThread.CurrentUICulture` for other threads?

I think this was it; all those strings in the wrong language were in background threads. Setting this in the latest commit:

https://docs.microsoft.com/en-us/dotnet/api/system.globalization.cultureinfo.defaultthreadcurrentuiculture?view=netframework-4.7.2

That was added in .NET Framework v4.5, which is our target framwork in the csproj, so it should be safe. No idea why it ever worked previously.

**Comment by DasSkelett** [comment]
> 
> 
> > Or we need to set `Thread.CurrentThread.CurrentUICulture` for other threads?
> 
> I think this was it; all those strings in the wrong language were in background threads. Setting this in the latest commit:
> 
> [docs.microsoft.com/en-us/dotnet/api/system.globalization.cultureinfo.defaultthreadcurrentuiculture?view=netframework-4.7.2](https://docs.microsoft.com/en-us/dotnet/api/system.globalization.cultureinfo.defaultthreadcurrentuiculture?view=netframework-4.7.2)
> 
> That was added in .NET Framework v4.5, which is our target framework in the csproj, so it should be safe. No idea why it ever worked previously.

Wow, this did indeed fix it. How weird. Thanks!

**Comment by vinix38** [comment]
random question here : do you think the issue you just fixed could also explain #3244 ?

**Comment by HebaruSan** [comment]
> random question here : do you think the issue you just fixed could also explain #3244 ?

Good question. I wish that was the case, but I don't think it is. I don't see a path by which an unexpected language could make it into either variable; at worst, it should revert to your OS level default, which no one reporting that issue has observed.

**Comment by DasSkelett** [comment]
Removed a few image resource imports from pt-BR and fr-FR, which reduces ckan.exe binary size back down to 3.4 MiB instead of increasing it to 4.3 MiB. The images aren't localized, so we are fine falling back to ones of the default language.

**Review by HebaruSan — APPROVED** [review]
Looks good to me, too. Thanks for the work, @gsantos9489 and @vinix38!

And thanks @DasSkelett for the review. If you wouldn't mind handling the merge, I'm in Windows at the moment where my Python can't run the merge script.

**Pull request merged**